src/AvaloniaEdit/Editing/LineNumberMargin.cs

```diff
@@ -108,6 +123,13 @@ private void OnDocumentLineCountChanged(object sender, EventArgs e)
             OnDocumentLineCountChanged();
         }
 
+        void TextViewVisualLinesChanged(object sender, EventArgs e)
+        {
+            InvalidateVisual();
+            InvalidateMeasure();
```

Comment:
`InvalidateVisual()` allone was not enough to make it work.

Reply:
What about invalidate measure by itself?

Reply:
for this exact scenario with font size it works